### PR TITLE
Nylas Availability: List view

### PR DESCRIPTION
**Pull request by philrenaud**
- Introduces a new `view_as` property with options of `schedule` (default) or `list`
- List view shows buttons consecutively without any height-based availability or styling. Hides all busy times.
- Also adds the properties to `<nylas-schedule-editor>` for manifest manipulation later

CleanShot 2021-09-20 at 16 52 00

# License

I confirm that this contribution is made under the terms of the MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Comment by Tabbbbs** [comment]
> This pull request has been linked to [Shortcut Story #69335: Nylas Availability: List-item view (for mobile)](https://app.shortcut.com/nylas/story/69335/nylas-availability-listitem-view-for-mobile).

File: components/availability/src/Availability.svelte [hunk 1]
```diff
@@ -938,6 +944,19 @@
...
+  function toggleSlot(slot: SelectableSlot) {
+    if (slot.selectionStatus === SelectionStatus.SELECTED) {
+      slot.selectionStatus = SelectionStatus.UNSELECTED;
+    } else {
+      if (slotSelection.length < max_bookable_slots) {
+        slot.selectionStatus = SelectionStatus.SELECTED;
+      }
+    }
+    days = [...days];
+  }
```
Comment (heathergh) [inline components/availability/src/Availability.svelte:958]: Just curious why this isn't
```
function toggleSlot(slot: SelectableSlot) {
  if (slot.selectionStatus === SelectionStatus.SELECTED) {
    slot.selectionStatus = SelectionStatus.UNSELECTED;
  } else if (slotSelection.length < max_bookable_slots) {
    slot.selectionStatus = SelectionStatus.SELECTED;
  }
  days = [...days];
}
```

File: components/availability/src/Availability.svelte [hunk 2]
```diff
@@ -1145,6 +1182,44 @@
...
+          }
+
+          &.selected {
+            background: purple;
+            color: white;
+          }
+
+          .partial {
+            display: block;
+            font-size: 0.6rem;
```
Comment (heathergh) [inline components/availability/src/Availability.svelte:1219]: what is this, a font-size for ants?! 😝

Comment (philrenaud) [reply components/availability/src/Availability.svelte:1219]: It's tiny but I think any bigger and I'd opt to just not have it -- it really muddies up the visuals!

File: components/availability/src/Availability.svelte [hunk 3]
```diff
@@ -938,6 +944,19 @@
     days = [...days]; // re-render
     resetDragState();
   }
+
+  // Click action for un-draggable slot-list buttons (list view)
+  function toggleSlot(slot: SelectableSlot) {
+    if (slot.selectionStatus === SelectionStatus.SELECTED) {
+      slot.selectionStatus = SelectionStatus.UNSELECTED;
+    } else {
+      if (slotSelection.length < max_bookable_slots) {
```
Comment (heathergh) [inline components/availability/src/Availability.svelte:953]: Just curious why this isn't `else if (slotSelection.length < max_bookable_slots) {...}`

**Review by heathergh — APPROVED** [review]

**Review by Tabbbbs — APPROVED** [review]

**Review by chantalala — APPROVED** [review]
Thanks @philrenaud, this looks great! As discussed on Slack, we can take care of the selection color in a future PR.

File: components/availability/src/Availability.svelte [hunk 4]
```diff
@@ -1305,73 +1385,109 @@
...
+        {:else if view_as === "list"}
+          <div class="slot-list">
+            {#each day.slots.filter((slot) => slot.availability === AvailabilityStatus.FREE || slot.availability === AvailabilityStatus.PARTIAL) as slot}
+              <button
+                data-available-calendars={slot.available_calendars.toString()}
+                aria-label="{new Date(
+                  slot.start_time,
+                ).toLocaleString()} to {new Date(
+                  slot.end_time,
+                ).toLocaleString()}; Free calendars: {slot.available_calendars.toString()}"
```
Comment (AaronDDM) [inline components/availability/src/Availability.svelte:1467]: Does this actually work? With the ""?
```suggestion
                aria-label={new Date(
                  slot.start_time,
                ).toLocaleString()} to {new Date(
                  slot.end_time,
                ).toLocaleString()}; Free calendars: {slot.available_calendars.toString()}
```

**Review by AaronDDM — APPROVED** [review]

File: components/availability/src/Availability.svelte (hunk 4 above)
Comment (philrenaud) [reply components/availability/src/Availability.svelte:1467]: Yep, and it's necessary as well here, because of the `...} to {...` bit!

**Pull request merged**